src/ert/substitutions.py

```diff
@@ -35,7 +35,11 @@ def substitute_parameters(
     ) -> str:
         for values in data.values():
             for key, value in values.items():
-                to_substitute = to_substitute.replace(f"<{key}>", f"{value:.6g}")
+                if isinstance(value, (int, float)):
```

Comment:
The typing should be updated to reflect the possibility to submit strings to this function.